File: Softteer/성적평가.py
n = int(input())
finalScroe = {i: 0 for i in range(n)}

for _ in range(3):
    rank = {i: 1 for i in range(n)}
    score = list(map(int, input().split()))

    for i in range(n):
        finalScroe[i] += score[i]
        for j in range(n):
            if score[j] > score[i]:
                rank[i] += 1
    print(' '.join([str(s) for s in rank.values()]))

finalRank = {i: 1 for i in range(n)}

for i in range(n):
    finalRank[i] = 1
    for j in range(n):
        if finalScroe[j] > finalScroe[i]:
            finalRank[i] += 1

print(' '.join([str(s) for s in finalRank.values()]))

File: Softteer/성적평가.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n'); //줄바꿈 처리 추가

const n = input[0];
const scores = input.slice(1);

let print = '';

const sums = Array(Number(n)).fill(0);

for (let score of scores) {
  const mems = score.split(' ').map(Number);
  const sorted = [...mems].sort((a, b) => b - a);

  for (let i in mems) {
    print += sorted.indexOf(mems[i]) + 1 + ' ';
    sums[i] += mems[i];
  }
  print += '\n';
}

const sorted = [...sums].sort((a, b) => b - a);
for (let s of sums) {
  print += sorted.indexOf(s) + 1 + ' ';
}

console.log(print);
